buildSrc/src/main/kotlin/toolkit-detekt.gradle.kts

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import io.gitlab.arturbosch.detekt.Detekt
+import io.gitlab.arturbosch.detekt.DetektCreateBaselineTask
+
+val detektVersion: String by project
+
+plugins {
+    id("io.gitlab.arturbosch.detekt")
+    id("toolkit-testing")
+}
+
+dependencies {
+    detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:$detektVersion")
+    detektPlugins(project(":detekt-rules"))
+}
+
+detekt {
+    val rulesProject = project(":detekt-rules").projectDir
+    input.from("$projectDir")
+    buildUponDefaultConfig = false
+    parallel = true
+    allRules = false
+    config = files("$rulesProject/detekt.yml")
+    baseline = file("$rulesProject/baseline.xml")
+
+    reports {
+        html.enabled = true // observe findings in your browser with structure and code snippets
+        xml.enabled = true // checkstyle like format mainly for integrations like Jenkins
+        sarif.enabled = true // standardized SARIF format to support integrations with Github Code Scanning
```

Comment:
do we need to upload the SARIF file?

https://docs.github.com/en/code-security/secure-coding/uploading-a-sarif-file-to-github#uploading-a-code-scanning-analysis-with-github-actions